addons/mail/models/mail_message.py

```diff
@@ -1016,13 +1016,19 @@ def _message_format(self, fnames):
     def message_fetch_failed(self):
         """Returns first 100 messages, sent by the current user, that have
         errors, in the format expected by the web client."""
-        messages = self.search([
-            ('has_error', '=', True),
-            ('author_id', '=', self.env.user.partner_id.id),
-            ('res_id', '!=', 0),
-            ('model', '!=', False),
-            ('message_type', '!=', 'user_notification')
-        ], limit=100)
+        author_id = self.env.user.partner_id.id
+        query = """
+                SELECT m.id from mail_message m
+                JOIN mail_message_res_partner_needaction_rel ON m.id = mail_message_res_partner_needaction_rel.mail_message_id
+                AND mail_message_res_partner_needaction_rel.notification_status in ('bounce','exception')
```

Comment:
This where condition is coming from a search method on a non-stored computed field `has_error`. Turning that into a static where condition is a bit error-prone if the search method gets overriden.